### ISSUE TITLE
Test UDF functions are revealed

### ISSUE THREAD

**Issue by MichaelDrogalis** [confluentinc/ksql#3094]
When I run `describe functions`, I get test UDFs like `TEST_UDF` and `COMPLEXFUNCTION`. These shouldn't be visible.

**Comment by stevenpyzhang** [comment]
This is caused by https://github.com/confluentinc/ksql/issues/2804 . Will close this issue for now and hold off on doing any further work until there's further discussion for the other issue. There are some quick solutions that can address this, but it doesn't make sense to do them if resolving #2804 will fix this bug anyways.

I verified between the 5.2.x and 5.3.x branch that the functions only show up in the 5.3.x branch, which is when ksql-functional-tests were added. In total there are 6 functions that began showing up:

 COMPLEXFUNCTION  
 CONFIGURABLEUDF       
 E2ECONFIGURABLEUDF   
 SOMEFUNCTION      
 TEST_UDAF      
 TEST_UDF    

The reason why they show up is that in UdfLoader.java at line 140, fastClasspathScanner scans all jar files in the directory that contain "ksql-engine" in the path name so that it can find public internal UDF and UDAF to add to the functionRegistry. In ksql-functional-tests, there's a copy of the ksql-engine test-jar (contains the test UDF/UDAF) since that's a dependency for the functional tests. The scanner picks up on the test-jar, which results in the test functions being added to the function registry.

ksql-engine-5.4.0-SNAPSHOT-tests.jar
